[16.0][MIG] shopinvader_image

lmignon opened the pull request:
* [x] depends on #1390
* [x] depends on https://github.com/OCA/search-engine/pull/158
* [x] depends on https://github.com/OCA/search-engine/pull/159

marielejeune commented on shopinvader_search_engine_image/tests/common.py:
```diff
@@ -0,0 +1,187 @@
+            {"name": "Fake SE", "tech_name": "fake_se", "backend_type": "fake"}
+        )
+
+        # create indexes for categories and products
+        cls.env["se.index"].create(
+            {
+                "name": "product",
+                "backend_id": cls.backend.id,
+                "model_id": cls.env.ref("product.model_product_product").id,
+                "serializer_type": "shopinvader_product_exports",
+            }
+        )
+        cls.env["se.index"].create(
+            {
+                "name": "category",
+                "backend_id": cls.backend.id,
+                "model_id": cls.env.ref("product.model_product_category").id,
+                "serializer_type": "shopinvader_category_exports",
+            }
+        )
```
```suggestion
```
Duplicate with lines 92-107

marielejeune commented on shopinvader_search_engine_image/schemas/shopinvader_category.py:
```diff
@@ -0,0 +1,20 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.shopinvader_product.schemas.category import (
+    ShopinvaderCategory as BaseShopinvaderCategory,
```
```suggestion
    ShortShopinvaderCategory as BaseShopinvaderCategory,
```
?

marielejeune commented on shopinvader_search_engine_image/tests/__init__.py:
```diff
@@ -0,0 +1 @@
+from . import test_shopinvader_search_engine_image
```
```suggestion
from . import test_shopinvader_search_engine_image
from . import test_images
```

lmignon replied:
I still need to cleanup the tests and `test_images` is the previous test file I need to rewrite.

marielejeune commented on shopinvader_search_engine_image/tests/common.py:
```diff
@@ -0,0 +1,184 @@
+                SeBackend,
+            )
+        )
+        cls.backend = cls.env["se.backend"].create(
+            {"name": "Fake SE", "tech_name": "fake_se", "backend_type": "fake"}
+        )
```
I don't know when tests ran on GitHub, but they fail for me today, when creating the backend:

```
 File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/addons/shopinvader_search_engine_image/tests/common.py", line 28, in setUpClass
    cls.backend = cls.env["se.backend"].create(
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/api.py", line 409, in _model_create_multi
    return create(self, [arg])
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/addons/server_environment/models/server_env_tech_name_mixin.py", line 64, in create
    return super().create(vals_list)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/addons/base/models/ir_fields.py", line 665, in create
    recs = super().create(vals_list)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/models.py", line 3967, in create
    next(iter(fields)).determine_inverse(batch_recs)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/fields.py", line 1377, in determine_inverse
    determine(self.inverse, records)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/fields.py", line 96, in determine
    if needle.__name__.find('__'):
AttributeError: 'functools.partial' object has no attribute '__name__'

```

lmignon committed:
[MIG] shopinvader_search_engine_image: Migration from 14.0

To improve naming consistency the original shopinvader_image addon is
renamed shopinvader_search_engine_image

qgroulard commented on shopinvader_search_engine_image/schemas/product_category.py:
```diff
@@ -0,0 +1,14 @@
+class ProductCategory(BaseCategory, ImageMixin, extends=True):
+    @classmethod
+    def from_product_category(cls, odoo_rec):
+        obj = super().from_product_category(odoo_rec, with_parent=True, with_child=True)
```
@lmignon Here is the bug on category bindings.

```suggestion
        obj = super().from_product_category(odoo_rec)
```

lmignon replied:
Thank you @qgroulard

lmignon commented:
/ocabot merge nobump
